Gudden/src/model/Indexer.java

```diff
@@ -1,5 +1,41 @@
 package model;
 
-public class Indexer {
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
 
+public class Indexer {
+	
+	private HashMap<String, List<PositionalPosting>> mIndex;
+	   
+	   
+	   public Indexer() {
+	      mIndex = new HashMap<String, List<PositionalPosting>>();
+	   }
+	   
+	   public void addTerm(String term, int documentID, int position) {
+	      if(!mIndex.containsKey(term))
+	    	  mIndex.put(term, new ArrayList<PositionalPosting>());
+	      List<PositionalPosting> docList = mIndex.get(term);
+	      if((docList.size()==0) || (docList.get(docList.size()-1).getDocId() < documentID) )
```

Comment:
You can use isEmpty instead of size == 0 and what do you mean ```(docList.get(docList.size()-1).getDocId() < documentID) ```? You can create a function for that.